src/Compilers/CSharp/Portable/BoundTree/VariablePendingInference.cs

```diff
@@ -79,8 +79,12 @@ internal BoundExpression SetInferredType(TypeSymbol type, Binder binderOpt, Diag
 
                     return new BoundFieldAccess(this.Syntax,
                                                 this.ReceiverOpt,
-                                                fieldSymbol, null, LookupResultKind.Viable, type,
-                                                this.HasErrors || inferenceFailed);
+                                                fieldSymbol,
+                                                null,
+                                                LookupResultKind.Viable,
+                                                isDeclaration: true,
```

Comment:
why is this a declaratoin?

Reply:
VariablePendingInference is only ever used in a `var x = ...` scenario. It's always a declaration.

Reply:
wait... which are the ones that are declarations, and which are the ones that aren't?

Reply:
Not sure I understand what you mean? Any usage of VariablePendingInference is a declaration. See https://github.com/dotnet/roslyn/pull/21643/files#diff-43badc526d6d686eda4eb66c4c7a8127R66, which is the case when the VariablePendingInference is a local reference, not a field reference.